### PR TITLE
Adding property files option in the Spark Submit command

### LINKED ISSUE #34838: Adding property files option in the Spark Submit command
### Description

spark-submit command has one of the options to pass the properties file as argument. Instead of loading multiple key, value via --conf option, this will help to load extra properties from the file path. While we have support for most of the arguments supported in spark-submit command in SparkSubmitOperator, this one `property-files` is missing. Could that be included?
```[root@airflow ~]# spark-submit --help
Usage: spark-submit [options] <app jar | python file | R file> [app arguments]
Usage: spark-submit --kill [submission ID] --master [spark://...]
Usage: spark-submit --status [submission ID] --master [spark://...]
Usage: spark-submit run-example [options] example-class [example args]

Options:

 --conf, -c PROP=VALUE       Arbitrary Spark configuration property.
  --properties-file FILE      Path to a file from which to load extra properties. If not
                              specified, this will look for conf/spark-defaults.conf.
```

### Use case/motivation

Add the property-files as one of the options to pass in the SparkSubmitOperator to load the extra config properties as file

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by ghostp13409**
---
Hi @Bowrna 

closes: #34838 

I've added property-files operator in the spark submit operator.
As I'm a newcomer to this industry I'm not sure if I need to do any other changes other than this. please let me know if there is a need to do any additional changes.

### CODE REVIEW

**Comment by eladkal** [comment]
can this be tested with unit test?

**Comment by ghostp13409** [comment]
> can this be tested with unit test?

Hi @eladkal,

As I mentioned above, I am a newcomer to open source and a also the development as a whole. So I haven't done any testing yet. However I would love to do so if you can guide me on how to do it.

File: airflow/providers/apache/spark/operators/spark_submit.py [hunk 1]
```diff
@@ -160,6 +160,11 @@ def on_kill(self) -> None:
             self._hook = self._get_hook()
         self._hook.on_kill()
 
+    def property_files(self) -> None:
+        if self.hook is None:
+            self._hook = self._get_hook()
+        self._hook.property_files()
+
     def _get_hook(self) -> SparkSubmitHook:
```
Comment (Bowrna) [inline airflow/providers/apache/spark/operators/spark_submit.py:166]: you have to accept the property_files param value as one of the parameters that can be used when creating SparkSubmitOperator. This code doesn't involve those changes.

**Comment by Bowrna** [comment]
> > can this be tested with unit test?
> 
> Hi @eladkal,
> 
> As I mentioned above, I am a newcomer to open source and a also the development as a whole. So I haven't done any testing yet. However I would love to do so if you can guide me on how to do it.

Please check if you can add relevant test in the test_spark_submit.py file for the changes you have done in spark_submit.py file. @ghostp13409

File: airflow/providers/apache/spark/operators/spark_submit.py (hunk 1 above)
Comment (ghostp13409) [reply airflow/providers/apache/spark/operators/spark_submit.py:166]: Okay will try to add that as well

Comment (ghostp13409) [reply airflow/providers/apache/spark/operators/spark_submit.py:166]: I've added properties-file param to Operator and Hook files. However, I haven't added the default file path handling logic as it seems to be already handled by `spark-submit` command as per the [docs](https://books.japila.pl/apache-spark-internals/tools/spark-submit/#driver-cores). I will follow up with the test case changes in the next commit. let me know if there is any feedback or concern.

> you have to accept the property_files param value as one of the parameters that can be used when creating SparkSubmitOperator. This code doesn't involve those changes.

**Comment by Bowrna** [comment]
hi @ghostp13409 any update on this? if you are struck in this let me know... i will see if i can support you. also some static checks are failing. you may need to fix those parts too.

**Comment by ghostp13409** [comment]
> hi @ghostp13409 any update on this? if you are struck in this let me know... i will see if i can support you. also some static checks are failing. you may need to fix those parts too.

HI @Bowrna,

thanks for reaching out.

A possible way to make the text more professional is:

I am working on setting up the dev environment, but I am encountering an error while installing breeze. I have tried different platforms (Windows and Linux) and versions of python and pyenv, but the error persists. I would appreciate your guidance on how to resolve this issue, as I have already asked about it on the slack community.

**Comment by Bowrna** [comment]
can you point out where you posted the question? @ghostp13409

**Comment by ghostp13409** [comment]
> can you point out where you posted the question? @ghostp13409

I posted my question on [#troubleshooting](https://apache-airflow.slack.com/archives/CCQ7EGB1P/p1697895028537189) and [#random](https://apache-airflow.slack.com/archives/CCR6P6JRL/p1697111516154339).

**Comment by pateash** [comment]
Hi,
I have a related issue #35911,
@ghostp13409 if you are stuck, I can pull your branch into mine and merge both changes and raise a PR.
let me know if that works.
cc. @eladkal @Bowrna

**Comment by ghostp13409** [comment]
yeah sure, that would be very helpful.
________________________________

**Comment by Bowrna** [comment]
@pateash @ghostp13409 if you need any help to push the fix for this issue, let me know.

**Comment by ghostp13409** [comment]
> https://github.com/apache/airflow/pull/36164
> will be replacing  this PR
> @ghostp13409 I have pulled your commits in, so your contribution will be counted as well.
> we can close this PR.

@pateash thanks

**Pull request closed without merging**